builder/slide_and_fold_storage.cc

```diff
@@ -264,12 +264,12 @@ class slide_and_fold : public node_mutator {
     // is produced there.
     if (loops.size() < 2 || !is_produced_by(output, body)) return;
 
-    auto ff = fold_factors[output];
     bool did_overlapped_fold = false;
 
-    if (ff) {
-      for (int d = 0; d < static_cast<int>(ff->size()); ++d) {
-        expr overlap = (*ff)[d].second;
+    if (fold_factors[output]) {
+      for (int d = 0; d < static_cast<int>(fold_factors[output]->size()); ++d) {
+        expr overlap = (*fold_factors[output])[d].second;
+        // TODO(vksnk): this is a bug, overlap is now always defined.
```

Comment:
Should it be `is_finite` instead?

Reply:
Fixed, it should be is_finite and also check that fold_factor is not the same as overlap.